src/main/java/space/obminyashka/items_exchange/rest/UserController.java

```diff
@@ -165,11 +166,19 @@ public ResponseEntity<String> updateUserPassword(@Valid @RequestBody ChangePassw
     @Operation(summary = "reset user password")
     @ApiResponses(value = {
             @ApiResponse(responseCode = "200", description = "OK"),
-            @ApiResponse(responseCode = "400", description = "BAD REQUEST")
-    })
-    public String resetPassword(@Valid @RequestBody ValidatedEmailRequest validatedEmailRequest){
-        mailService.sendMessagesToEmailForResetPassword(validatedEmailRequest.email());
-        return getMessageSource(ResponseMessagesHandler.PositiveMessage.RESET_PASSWORD);
+            @ApiResponse(responseCode = "400", description = "BAD REQUEST"),
+            @ApiResponse(responseCode = "404", description = "NOT_FOUND")})
+    @ResponseStatus(HttpStatus.OK)
+    public String resetPassword(@Valid @RequestBody ValidatedEmailRequest validatedEmailRequest,
+                                @Parameter(hidden = true) @RequestHeader(HttpHeaders.HOST) String host) {
+        var email = validatedEmailRequest.email();
+        var user = getUser(email);
+        if (userService.existsByEmail(email)) {
```

Comment:
You can remove this if because getUser throw exception in case when user by email not found

Reply:
Valid point and it seems that EntityNotFoundException is also excess.

Reply:
removed